### PR TITLE
Adding tag management through url #22

### PR DESCRIPTION
**Pull request by sirish123**

### CODE REVIEW

**Comment by Rahilsiddique** [comment]
when visiting the the `tags` page in the preview, website seems to break in prod.

**Comment by sirish123** [comment]
Any reason why?

**Comment by sirish123** [comment]
Could u check the functionality on your local  or give me the logs of the deployment and tell me where the error could be

**Comment by Rahilsiddique** [comment]
these are the logs from vercel when I try to visit the `tags`  route 
```

 ⚠ metadata.metadataBase is not set for resolving social open graph or twitter images, using "https://next-blog-pvhy5b55w-rahilsiddique.vercel.app". See https://nextjs.org/docs/app/api-reference/functions/generate-metadata#metadatabase
Error: ENOENT: no such file or directory, scandir 'src/posts'
    at Object.readdirSync (node:fs:1527:3)
    at o (/var/task/.next/server/app/tags/page.js:1:3228)
    at l (/var/task/.next/server/app/tags/page.js:1:2486)
    at ev (/var/task/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:109:11059)
    at /var/task/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:109:21907
    at Array.toJSON (/var/task/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:109:25197)
    at stringify (<anonymous>)
    at eP (/var/task/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:109:13686)
    at e$ (/var/task/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:109:14212)
    at Timeout._onTimeout (/var/task/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:109:12034) {
  errno: -2,
  syscall: 'scandir',
  code: 'ENOENT',
  path: 'src/posts'
}

```

also sorting should be based on number of blogs and not alphabatical order as you are using the terms `asc` and `desc` which are used in the context of number thus may cause missunderstanding 

keep the logic same just instead of sorting alphabatically sort by number

**Comment by sirish123** [comment]
is there anything related to /tags specifically this does not seem to provide much info

**Comment by sirish123** [comment]
Three kinds of sorting added , alphabetical, number ascending and number descending, I cant seem to figure out what the error in prod is related to @Rahilsiddique if u want u check in to some other branch.

File: src/app/tags/page.tsx [hunk 1]
```diff
@@ -1,20 +1,52 @@
...
-    acc[curr] = (acc[curr] || 0) + 1;
-    return acc;
-  }, {});
+type searchProps = {
+  searchParams?: {
+    sort?: string;
+  };
+};
+
+export const dynamic = "force-dynamic";
```
Comment (Rahilsiddique) [inline src/app/tags/page.tsx]: myabe try removing this line, in build this page is rendered as serverless function but instead should be satitically generated like all other routes

Comment (sirish123) [reply src/app/tags/page.tsx]: still does not work

Comment (sirish123) [reply src/app/tags/page.tsx]: https://github.com/vercel/next.js/issues/43077 pls have a look at this i did everything related to the issue but problem seems to be the new /app directory

Comment (Rahilsiddique) [reply src/app/tags/page.tsx]: okk so i did a through invastigation, and everything you did seem to be correct one last thing you can still try is to add  ` (const sortType = searchParams?.sort || "alphabet")` circular brackets, if that also does not work just create a normal pr wihout the search param feature 
@sirish123  thanks 👍

**Comment by sirish123** [comment]
@Rahilsiddique the latest build its working but uses a "usestate" solution to bypass error we encountered, it uses CSR with an API call made to Server side I hope this is ok , the latest deployed build is working

**Review by Rahilsiddique — APPROVED** [review]
LGTM !!

**Pull request merged**